reconcile/external_resources/secrets_sync.py

```diff
@@ -99,40 +96,42 @@ def _init_ocmap(self, specs: Iterable[ExternalResourceSpec]) -> OCMap:
             integration=QONTRACT_INTEGRATION,
         )
 
-    @retry()
-    def _write_secrets_to_vault(self, spec: ExternalResourceSpec) -> None:
-        if spec.secret:
-            secret_path = f"{self.vault_path}/{QONTRACT_INTEGRATION}/{spec.cluster_name}/{spec.namespace_name}/{spec.output_resource_name}"
-            stringified_secret = {k: str(v) for k, v in spec.secret.items()}
-            desired_secret = {"path": secret_path, "data": stringified_secret}
-            self.vault_client.write(desired_secret, decode_base64=False)
-
     def sync_secrets(self, specs: Iterable[ExternalResourceSpec]) -> None:
         self._populate_secret_data(specs)
-        ri = ResourceInventory()
-        self._initialize_ri(ri, specs)
-        ocmap = self._init_ocmap(specs)
-        for item in ri:
-            self.reconcile_data(item, ri, ocmap)
+
+        to_sync_specs = [spec for spec in specs if spec.secret]
```

Comment:
this filter logic duplicated 3 times in this file, worth extracting a method to avoid errors when adding new logic.

Reply:
done.